src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -177,6 +177,110 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
         }
     };
 
+    template<typename T>
+    struct ConversionTrait<std::unordered_map<std::string, T>>
+    {
+        std::unordered_map<std::string, T> FromJson(const Json::Value& json) const
+        {
+            ConversionTrait<T> trait;
+            std::unordered_map<std::string, T> val;
+
+            for (const auto& element : json.getMemberNames())
+            {
+                val[element.c_str()] = trait.FromJson(json[JsonKey(element)]);

```

Comment:
You can reuse some of the other conversion magic here by doing...

It'll avoid copying the string again (and again).

```
GetValueForKey(json, element, val[element.c_str()]);
```